dracclient/resources/raid.py

```diff
@@ -98,15 +98,229 @@
+                                             'PossibleValues',
+                                             cls.namespace, find_all=True)]
+
+        return cls(raid_attr.name, raid_attr.instance_id,
+                   raid_attr.pending_value, raid_attr.read_only,
+                   raid_attr.fqdd, current_value, possible_values)
+
+    def validate(self, new_value):
+        """Validates new value"""
+
+        if str(new_value) not in self.possible_values:
+            msg = ("Attribute '%(attr)s' cannot be set to value '%(val)s'."
+                   " It must be in %(possible_values)r.") % {
+                       'attr': self.name,
+                       'val': new_value,
+                       'possible_values': self.possible_values}
+            return msg
+
+
+class RAIDStringAttribute(RAIDAttribute):
+    """String RAID attribute class"""
+
+    namespace = uris.DCIM_RAIDString
+
+    def __init__(self, name, instance_id, pending_value,
+                 read_only, fqdd, current_value, min_length, max_length):
+        """Creates RAIDStringAttribute object
+
+        :param name: name of the RAID attribute
+        :param instance_id: InstanceID of the RAID attribute
+        :param pending_value: pending value of the RAID attribute, reflecting
+                an unprocessed change (eg. config job not completed)
+        :param read_only: indicates whether this RAID attribute can be changed
+        :param fqdd: Fully Qualified Device Description of the RAID
+                Attribute
+        :param current_value: list containing the current values of the
+                RAID attribute
+        :param min_length: minimum length of the string
+        :param max_length: maximum length of the string
+        """
+        super(RAIDStringAttribute, self).__init__(name, instance_id,
+                                                  pending_value,
+                                                  read_only, fqdd)
+        self.current_value = current_value
+        self.min_length = min_length
+        self.max_length = max_length
+
+    @classmethod
+    def parse(cls, raid_attr_xml):
+        """Parses XML and creates RAIDStringAttribute object"""
+
+        raid_attr = RAIDAttribute.parse(cls.namespace, raid_attr_xml)
+        current_value = [attr.text for attr
+                         in utils.find_xml(raid_attr_xml,
+                                           'CurrentValue',
+                                           cls.namespace, find_all=True)]
+        min_length = int(utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'MinLength'))
+        max_length = int(utils.get_wsman_resource_attr(
+            raid_attr_xml, cls.namespace, 'MaxLength'))
+
+        return cls(raid_attr.name, raid_attr.instance_id,
+                   raid_attr.pending_value, raid_attr.read_only,
+                   raid_attr.fqdd, current_value, min_length, max_length)
+
```

Comment:
Can you add a RAIDInteger class and include this in the list() and set() methods?